robotpy_ext/misc/precise_delay.py

```diff
@@ -77,5 +77,47 @@ def _wait_unit_tests(self):
         # - TODO: should we just always use this in simulated mode?
         
         wpilib.Timer.delay(self.delay_period)
-        
-        
+
+
+class NotifierDelay:
+    """Synchronizes a timing loop against interrupts from the FPGA.
+
+    This will delay so that the next invocation of your loop happens at
+    precisely the same period, assuming that your loop does not take longer
+    than the specified period.
+
+    Please clean up after yourself when done using this object.
```

Comment:
You can use `Resource._add_global_resource` to have cleanup happen automatically in addition to the user calling free?

Reply:
Probably a good idea. Although having a dangling FPGA alarm sounds like a bad idea anyway. Hm.